### ISSUE TITLE
Allow to edit the Parent RA delegation info in the object page

### ISSUE THREAD

**Issue by danielmarschall** [danielmarschall/oidplus#52]

**Comment by danielmarschall** [comment]
WIP:

```diff
[00:38]oidplus@viathinksoft.de:~/hosted/dev$ svn diff
Index: TODO
===================================================================
--- TODO        (Revision 1482)
+++ TODO        (Arbeitskopie)
@@ -90,7 +90,6 @@
 - "Decoding" section (tech info) at OID nodes, e.g. the AID decoder, should this be a scrolling <pre> block instead of a <code> block? (For mobile devices)
 - problem: if an identical class file (with same namespace) is placed in two plugin folders, then the oidplus autoloader will include both, which will cause a fatal error
 - oidplus docker image?
-- In the object pages, there is now an delete button (requires parent write rights), but it would be also nice to edit ASN1/IRI/Comment there.
 
 Databases:
 - Let plugins create tables for all DBMS by including a "create table" function with abstract types in the SQL slang plugins (similar to Medoo)
Index: plugins/tushevorg/publicPages/2000_uitweaks/manifest.json
===================================================================
--- plugins/tushevorg/publicPages/2000_uitweaks/manifest.json   (Revision 1482)
+++ plugins/tushevorg/publicPages/2000_uitweaks/manifest.json   (Arbeitskopie)
@@ -6,7 +6,7 @@
                        "name": "UITweaks",
                        "author": "tushev.org",
                        "license": "MIT",
-                       "version": "1.1+viathinksoft20230323",
+                       "version": "1.1+viathinksoft20240615",
                        "descriptionHTML": "<a href=\"https://github.com/tushev/uitweaks-oidplus-plugin\" target=\"_blank\">UITweaks plugin for OIDplus 2.0</a>:\n\t\t<ul>\n\t\t<li>Fully expand Objects tree on page reload</li>\n\t\t<li>Collapse Login tree on page reload</li>\n\t\t<li>Collapse Documents &amp; Resources tree on page reload</li>\n\t\t<li>Change default tree pane width</li>\n\t\t<li>Remember tree pane width in <code>browser.localStorage</code> across page reloads</li>\n\t\t<li>Prefer \"Admin login\" tab at login page</li>\n\t\t</ul>",
                        "oid": "1.3.6.1.4.1.37476.9000.146.2.109.2000"
                },
@@ -18,4 +18,4 @@
                        "OIDplusPagePublicUITweaks.js"
                ]
        }
-}
\ No newline at end of file
+}
Index: plugins/viathinksoft/publicPages/000_objects/OIDplusPagePublicObjects.class.php
===================================================================
--- plugins/viathinksoft/publicPages/000_objects/OIDplusPagePublicObjects.class.php     (Revision 1482)
+++ plugins/viathinksoft/publicPages/000_objects/OIDplusPagePublicObjects.class.php     (Arbeitskopie)
@@ -1035,13 +1035,10 @@
 
                        $out['text'] = '<!-- MARKER 1 -->' . $out['text']; // use this to better control modifyContent!
 
-                       if ($objParent) {
-                               if ($objParent->userHasWriteRights()) {
-                                       $out['text'] = '<p style="float:right"><button type="button" class="btn btn-danger btn-xs delete" onclick="OIDplusPagePublicObjects.crudActionDelete('.js_escape($obj->nodeId(true)).', '.js_escape($objParent->nodeId(true)).')">'._L('Delete').'</button></p>' . $out['text'];
-                               }
+                       if ($objParent && $objParent->userHasWriteRights()) {
+                               $out['text'] = '<p style="float:right"><button type="button" class="btn btn-danger btn-xs delete" onclick="OIDplusPagePublicObjects.crudActionDelete('.js_escape($obj->nodeId(true)).', '.js_escape($objParent->nodeId(true)).')">'._L('Delete').'</button></p>' . $out['text'];
                        }
 
-
                        if ($objParent) {
                                if ($objParent->isRoot()) {
                                        $parent_link_text = $objParent->objectTypeTitle();
@@ -1115,6 +1112,31 @@
                                $out['text'] = str_replace('%%RA_INFO%%', OIDplusPagePublicRaInfo::showRaInfo($obj->getRaMail(), $obj), $out['text']);
                        }
 
+                       if ($objParent && $objParent->userHasWriteRights()) {
+                               // TODO: Make a bit nicer...
+                               $out['text'] .= '<h2>'._L('Superior RA Delegation').'</h2>';
+                               $out['text'] .= '<noscript><p><font color="red">'._L('You need to enable JavaScript to edit title or description of this object.').'</font></p></noscript>';
+                               $out['text'] .= '<div class="container box" style="display:none" id="suprabox">';
+                               $out['text'] .= _L('RA').': <input type="text" name="ra_email" id="suprabox_ra" value="'.htmlentities($obj->getRa()->raEMail()).'"><br><br>';
+                               $asn1ids = array();
+                               $res2 = OIDplus::db()->query("select name from ###asn1id where oid = ? order by lfd", array($obj->nodeId(true)));
+                               while ($row2 = $res2->fetch_array()) {
+                                       $asn1ids[] = $row2['name'];
+                               }
+                               $out['text'] .= _L('ASN.1 IDs (comma sep.)').': <input type="text" name="asn1ids" id="suprabox_asn1" value="'.htmlentities(implode(', ',$asn1ids)).'"><br><br>';
+                               $iris = array();
+                               $res2 = OIDplus::db()->query("select name from ###iri where oid = ? order by lfd", array($obj->nodeId(true)));
+                               while ($row2 = $res2->fetch_array()) {
+                                       $iris[] = $row2['name'];
+                               }
+                               $out['text'] .= _L('IRI IDs (comma sep.)').': <input type="text" name="iris" id="suprabox_iri" value="'.htmlentities(implode(', ', $iris)).'"><br><br>';
+                               $out['text'] .= _L('Comment').': <input type="text" name="comment" id="suprabox_comment" value="'.htmlentities($obj->getComment()).'"><br><br>';
+                               $out['text'] .= _L('Confidential').': <input type="checkbox" name="confidential" id="suprabox_hide" value="1"'.htmlentities($obj->isConfidential()?' checked':'').'><br><br>';
+                               $out['text'] .= '<button type="button" name="update_desc2" id="update_desc2" class="btn btn-success btn-xs update" onclick="OIDplusPagePublicObjects.updateDesc2('.js_escape($obj->nodeId()).')">'._L('Update').'</button>';
+                               $out['text'] .= '</div>';
+                               $out['text'] .= '<script>$("#suprabox")[0].style.display = "block";</script>';
+                       }
+
                        $out['text'] .= '<!-- MARKER 3 -->'; // use this to better control modifyContent!
                        $out['text'] .= '<!-- MARKER 4 -->'; // use this to better control modifyContent!
                        $out['text'] .= '<!-- MARKER 5 -->'; // use this to better control modifyContent!
Index: plugins/viathinksoft/publicPages/000_objects/OIDplusPagePublicObjects.js
===================================================================
--- plugins/viathinksoft/publicPages/000_objects/OIDplusPagePublicObjects.js    (Revision 1482)
+++ plugins/viathinksoft/publicPages/000_objects/OIDplusPagePublicObjects.js    (Arbeitskopie)
@@ -283,6 +283,50 @@
                });
        },
 
+       updateDesc2: function(id) {
+               $.ajax({
+                       url:"ajax.php",
+                       method:"POST",
+                       beforeSend: function(jqXHR, settings) {
+                               $.xhrPool.abortAll();
+                               $.xhrPool.add(jqXHR);
+                       },
+                       complete: function(jqXHR, text) {
+                               $.xhrPool.remove(jqXHR);
+                       },
+                       data: {
+                               csrf_token:csrf_token,
+                               plugin:OIDplusPagePublicObjects.oid,
+                               action:"Update",
+                               id:id,
+                               ra_email:($("#suprabox_ra")[0] ? $("#suprabox_ra")[0].value : null),
+                               asn1ids:($("#suprabox_asn1")[0] ? $("#suprabox_asn1")[0].value : null),
+                               iris:($("#suprabox_iri")[0] ? $("#suprabox_iri")[0].value : null),
+                               comment:($("#suprabox_comment")[0] ? $("#suprabox_comment")[0].value : null),
+                               confidential:($("#suprabox_hide")[0] ? $("#suprabox_hide")[0].checked : null)
+                       },
+                       error: oidplus_ajax_error,
+                       success: function (data) {
+                               oidplus_ajax_success(data, function (data) {
+                                       alertSuccess(_L("Update OK"));
+                                       reloadContent();
+                                       /*
+                                       $('#oidtree').jstree("refresh");
+                                       var h1s = $("h1");
+                                       for (var i = 0; i < h1s.length; i++) {
+                                               var h1 = h1s[i];
+                                               h1.innerHTML = $("#titleedit")[0].value.htmlentities();
+                                       }
+                                       document.title = combine_systemtitle_and_pagetitle(getOidPlusSystemTitle(), $("#titleedit")[0].value);
+
+                                       var mce = tinymce.get('description');
+                                       if (mce != null) mce.setDirty(false);
+                                       */
+                               });
+                       }
+               });
+       },
+
        frdl_weid_change: function() {
                var from_base = 36;
                var from_control = "#weid";
```

**Comment by danielmarschall** [comment]
Fixed in release 2.0.1.25
